### PR TITLE
Added newyear app to apps.loklak.org

### PR DESCRIPTION
**Pull request by rhemon**
Live: http://thawing-atoll-77925.herokuapp.com/apps/newyear/
Code for live: https://github.com/rhemon/loklak_server/tree/master (Should I do a PR for that repository as well?) 
Issue : https://github.com/fossasia/apps.loklak.org/issues/20
screen shot 2017-01-02 at 6 52 44 pm

I have a slight confusion on using `/api/search.json`, that adding it to apps.loklak.org wont run the app since there is no "/api", is there? and when I try using `api.loklak.org/api/search.json` I keep getting `Cross-Origin Request Blocked` , so instead I added the app to loklak_server, the way it was mentioned in boilerplate and made the live version out of that.

### CODE REVIEW

**Comment by mariobehling** [comment]
Please do not add apps to loklak_server anymore this is outdated. I am confused why some apps work and some do not. If this app does not work on apps.loklak.org, it cannot not be merged at this point. It does not make sense to merge apps, that do not work. 

Please work with @Orbiter to fix the CORS issue.

**Comment by Orbiter** [comment]
@rhemon all requests to the api must be done with jsonp now, like http://api.loklak.org/api/search.json?timezoneOffset=-60&q=linux&callback=p to overcome the same origin policy.

**Comment by rhemon** [comment]
@Orbiter So I have tried using jsonp and now getting this:
```
Loading mixed (insecure) active content "http://api.loklak.org/api/search.json?timezoneOffset=0&q=%23happy2017#happy2017&callback=p" on a secure page

SyntaxError: missing ; before statement search.json:2:12
```

My code: 
```
var app = angular.module("newyear", []);
app.controller("TweetsController", ["$scope", "$http", function($scope, $http) {
    $scope.d = new Date()
    $scope.n = $scope.d.getTimezoneOffset();  
    $scope.queryUrl = "http://api.loklak.org/api/search.json?timezoneOffset="+$scope.n.toString()+"&q=%23";
    $scope.queryTags = ["happynewyear", "newyear", "2017", "hello2017", "happy2017"];
    $scope.tweets = [];
    $scope.loadTweets = function() {
        for (var QIndex in $scope.queryTags) {
            var queryTag = $scope.queryTags[QIndex];
            console.log($scope.queryUrl+queryTag+"#"+queryTag+"&callback=p");
            $http.jsonp($scope.queryUrl+queryTag+"#"+queryTag+"&callback=p").success(function(data) {
                console.log("success");
                for (var SIndex in data.statuses){
                    
                    var status = data.statuses[SIndex];
                    if ($scope.tweets.indexOf(status) == -1) {
                        status.text = status.text.replace(/<span class=\"twitter-hashflag-container\">/g, '');
                        $scope.tweets.push(status);
                    }
                }
            }).error(function(data) {
                console.log("request failed");
            });
        }
    }
    $scope.loadTweets();
}])
```

**Comment by hemantjadon** [comment]
@rhemon this is the issue with `https` and `http`,
see this I have explained this in detail here https://github.com/fossasia/loklak_search/issues/146#issuecomment-269080080

**Comment by rhemon** [comment]
@hemantjadon I understand the https and http problem, but shouldn't it work once I tell it to load the script?

**Comment by rhemon** [comment]
@hemantjadon @Orbiter 
Not sure why, but if I set `callback` to `JSON_CALLBACK` I am able to get a response 
https://rhemon.github.io/apps.loklak.org/newyear/

screen shot 2017-01-04 at 2 02 39 pm

My js code : https://github.com/rhemon/apps.loklak.org/blob/master/newyear/js/newyear.js

**Comment by hemantjadon** [comment]
@rhemon 
This is is good stack overflow thread to understand what JSONP is. http://stackoverflow.com/questions/2067472/what-is-jsonp-all-about

Now what angular does while making JSONP request?

* Angular creates a `script tag` for you, and at last runs an internal callback function, thus you need to specify `?callback=JSONP_CALLBACK`, for proper results.

* The JSONP_CALLBACK simply resolves the promise, thus your code `.success()` runs after that.

In conclusion `?callback=JSONP_CALLBACK` is required for internal angular promise resolving.

**Comment by rhemon** [comment]
since the app now works can this be merged
https://rhemon.github.io/apps.loklak.org/newyear/ 
also my task link:
https://codein.withgoogle.com/dashboard/task-instances/5437375792545792/

**Comment by mariobehling** [comment]
Thank you!
